react/week4/managing-state/src/index.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import * as serviceWorker from './serviceWorker';
+
+// Importing components
+import Context from './context';
+import Searchuser from './searchuser';
+import ListItems from './userListItems';
+
+// Context
+const Contextprovider = Context.Provider;
+
+function FetchArray(event){
+  const URL=`https://api.github.com/search/users?q=${event}`;
+    return fetch(URL).then(response => response.json())        
+}
+
+class TodoList extends React.Component{
+  state = {
+    isLoading: false,
+    Todo: [],
+  }
+
+  fetchlist=(event) => {    
+    FetchArray(event).then(data => {     
+      this.setState({ Todo: data.items})
+    }).catch((error) => {
+      if(error){
+      alert(`This Process Failed : ${error}`);
+      return
+      }
+    }).finally(() => {
+      this.setState({isLoading:false})
+    })
+  }
+
+  handleChange=(event) => {
+    const inputevent = event.target.value;
+    if(!inputevent){
+      return
+    }
+    this.setState({isLoading:true})
```

Comment:
I would move this inside the `fetchlist` method instead. This way you can always be guaranteed the state will be changed as you expect. In this setup, if somewhere were to call handleChange without fetchlist, you'd be stuck loading forever.